### PR TITLE
Replace outdated react-firebaseui package

### PR DESCRIPTION
**Pull request by MvRemmerden**
This replaces the outdated `react-firebaseui` package as mentioned in https://github.com/gladly-team/next-firebase-auth/issues/555.

### CODE REVIEW

File: example/components/StyledFirebaseAuth.tsx
Comment (kmjennison) [inline example/components/StyledFirebaseAuth.tsx:1]: Please make small edits here to satisfy linting rules.

File: example/package-lock.json
Comment (kmjennison) [inline example/package-lock.json:1]: Please delete package-lock.json in favor of yarn.lock.

**Review by kmjennison — CHANGES_REQUESTED** [review]
Thank you for this contribution! It looks great. Just requesting a few quick edits.

File: example/package-lock.json
Comment (MvRemmerden) [reply example/package-lock.json:1]: Done!

File: example/components/StyledFirebaseAuth.tsx
Comment (MvRemmerden) [reply example/components/StyledFirebaseAuth.tsx:1]: Done!

**Comment by MvRemmerden** [comment]
> Thank you for this contribution! It looks great. Just requesting a few quick edits.

Done, back to you 🙂

File: package.json [hunk 1]
```diff
@@ -132,6 +132,7 @@
   "dependencies": {
     "@babel/runtime": "^7.23.7",
     "cookies": "^0.8.0",
+    "firebaseui": "^6.1.0",
```
Comment (kmjennison) [inline package.json]: This isn't a dependency of the main package, just the example app, so let's remove this.
```suggestion
    "firebaseui": "^6.1.0",
```

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: example/components/StyledFirebaseAuth.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  // See: https://github.com/firebase/firebaseui-web#configuration
+  uiConfig: auth.Config
+  // Callback that will be passed the FirebaseUi instance before it is
+  // started. This allows access to certain configuration options such as
+  // disableAutoSignIn().
+  uiCallback?(ui: auth.AuthUI): void
+  // The Firebase App auth instance to use.
+  firebaseAuth: Auth // As firebaseui-web
+  className?: string
+}
```
Comment (kmjennison) [inline example/components/StyledFirebaseAuth.tsx:17]: This lint rule is no longer relevant. Could you add `"react/require-default-props": "off"` to the .eslintrc.js rules?

**Review by kmjennison — CHANGES_REQUESTED** [review]
Just requesting another minor lint fix. Reminder that you can check linting yourself locally by running `yarn lint` at the package root. Thanks for these edits!

File: example/components/StyledFirebaseAuth.tsx (hunk 2 above)
Comment (MvRemmerden) [reply example/components/StyledFirebaseAuth.tsx:17]: Done!

> Reminder that you can check linting yourself locally by running yarn lint at the package root.

Thanks, I wasn't aware. I'm usually not working with `eslint`, so this was mostly trial & error.

**Review by kmjennison — APPROVED** [review]
Looks great. Thank you for the contribution and revisions!

**Comment by MvRemmerden** [comment]
> Looks great. Thank you for the contribution and revisions!

No worries, thanks for the guidance! Very happy to have contributed at least something back to this project, which has been super helpful for me! 🙂

**Pull request merged**